File: Java/PostorderTraversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {

    public List<Integer> postorderIterative(TreeNode root) {
        LinkedList<Integer> res = new LinkedList<>();
        if(root == null) return res;
        Stack<TreeNode> stack = new Stack<>();
        stack.push(root);

        while(!stack.isEmpty()) {
            root = stack.pop();
            res.addFirst(root.val);
            if(root.left != null) stack.push(root.left);
            if(root.right != null) stack.push(root.right);
        }

        return res;
    }

    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        if(root == null) return res;

        List<Integer> left = postorderTraversal(root.left);
        List<Integer> right = postorderTraversal(root.right);
        if(left != null && left.size() > 0) res.addAll(left);
        if(right != null && right.size() > 0) res.addAll(right);
        res.add(root.val);

        return res;
    }
}

File: Go/postorder_traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func postorderTraversal(root *TreeNode) []int {
    if root == nil {
        return []int{}
    }
    
    var stack []*TreeNode
    var res []int 
    l := list.New()
    stack = append(stack, root)

    for len(stack) > 0 {
        n := stack[len(stack) - 1]
        l.PushFront(n.Val)
        stack = stack[:len(stack) - 1]
        if n.Left != nil {
            stack = append(stack, n.Left)
        }
        if n.Right != nil {
            stack = append(stack, n.Right)
        }
    }

    for p := l.Front(); p != nil; p = p.Next() {
        res = append(res, p.Value.(int))
    }

    return res
}